### PR TITLE
ci: Add test support for external database adapter

### LINKED ISSUE #8788: Create dummy DB adapter repo that runs Parse Sever tests
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Feature / Enhancement Description

This purpose of this issue is to centralize the discussion around how to create a dummy DB adapter repo as a POC to demonstrate the [test exclusion list](https://github.com/parse-community/parse-server/issues/8714) feature while running the full Parse Server test suite externally.

It should probably work like so:

1. The adapter repo CI downloads the full Parse Server repo directly from GitHub based on a version tag. To ensure the adapter is compatible with specific versions, the CI could use a matrix to download multiple Parse Server major versions, e.g. the latest 5.x and 6.x. Downloading from npm wouldn't contain the tests as they are stripped from the published package.
2. The CI modifies the DB adapter that Parse Server uses to run the tests. This may require some login the Parse Server repo  test helper logic.
3. The CI overrides the `textExclusionList.json` in Parse Server with the file defined in the adapter repo. So that Parse Server will exclude these tests. Or maybe more elegant, add an optional path param to the npm test command, so the CI just starts the CI by passing the file path.

### PR DESCRIPTION
**Pull request by ddrechse**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Closes: https://github.com/parse-community/parse-server/issues/8788

## Approach
 
Add support to dynamically import an external database adapter to run tests.

## Tasks
n/a

### CODE REVIEW

File: spec/helper.js [hunk 1]
```diff
@@ -444,10 +448,8 @@ global.it_id = (id, func) => {
   if (testExclusionList.includes(id)) {
     return xit;
   } else {
-    if(func === undefined)
-      return it;
-    else
-      return func;
+    if (func === undefined) return it;
+    else return func;
```
Comment (mtrezza) [inline spec/helper.js]:
```suggestion
    return func || it;
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: spec/helper.js (hunk 1 above)
Comment (ddrechse) [reply spec/helper.js]: Merged

**Comment by mtrezza** [comment]
@ddrechse Since I guess we cannot easily add a test for this; do you have an example that shows that this works?

**Comment by ddrechse** [comment]
I do
I have added the adapter support to the repo here
https://github.com/oracle-samples/oracleadapter-parse

This is an example config.json for the Oracle Storage Adapter running against Free23c locally
```
{
  "appId": "APPLICATION_ID",
  "masterKey": "MASTER_KEY",
  "port": 1338,
  "logLevel": "info",
  "verbose": false,
  "mountGraphQL": true,
  "mountPlayground": true,
  "graphQLPath": "/graphql",
  "filesAdapter": {
    "module": "@parse/fs-files-adapter"
  },
  "databaseAdapter": {
    "module": "../Adapters/Storage/Oracle/OracleStorageAdapter",
    "options": {
      "databaseURI": "oracledb://pdbadmin:Welcome12345@localhost:1521/freepdb1",
      "collectionPrefix": ""
    }
  }
}  

```
@mtrezza

**Comment by mtrezza** [comment]
Could you test this out by using the branch of this PR in https://github.com/parse-community/parse-server-database-adapter-template/pull/4? Looking at the CI there, it's using a Parse Server version tag when installing Parse Server, so it doesn't have the feature of this PR yet and the CI there actually runs with the MongoDB storage adapter, right? If you could change it to use the branch of this PR, then passing the env vars in the CI should run the CI with Oracle DB adapter, and once we see that works, we can merge this PR, does that make sense?

**Comment by ddrechse** [comment]
Yes it does

**Comment by dplewis** [comment]
@mtrezza @ddrechse is this ready to merge? I have a MySQL adapter that can use this. MySQL and PostgreSQL are similar so moving PG to another repo would be easier

**Comment by mtrezza** [comment]
@dplewis this is still ongoing development, partly outside of this org. Once we know it all works, we will merge and publish a repo template that you can use for your SQL adapter.

**Review by mtrezza — CHANGES_REQUESTED** [review]
Could you please check the coverage? Not sure if it's possible to write a simple test for this, otherwise we merge as is.

**Review by mtrezza — DISMISSED** [review]
Looks good; waiting for CI to pass...

File: src/cli/utils/runner.js [hunk 2]
```diff
@@ -5,12 +5,7 @@ function logStartupOptions(options) {
     return;
   }
   // Keys that may include sensitive information that will be redacted in logs
-  const keysToRedact = [
-    'databaseURI',
-    'masterKey',
-    'maintenanceKey',
-    'push',
-  ];
+  const keysToRedact = ['databaseAdapter', 'databaseURI', 'masterKey', 'maintenanceKey', 'push'];
```
Comment (mtrezza) [inline src/cli/utils/runner.js]:
```suggestion
  const keysToRedact = [
    'databaseAdapter',
    'databaseURI',
    'masterKey',
    'maintenanceKey',
    'push',
  ];
```

**Review by mtrezza — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.1.0-alpha.7](https://github.com/parse-community/parse-server/releases/tag/7.1.0-alpha.7)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.1.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.1.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.1.0](https://github.com/parse-community/parse-server/releases/tag/7.1.0)

**Pull request merged**